app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,35 @@
+class SessionsController < Devise::SessionsController
+  layout 'empty', :only => [ :timeout_check ]
+
+  def timeout_check
+    return if !session || !session["warden.user.user.session"] # session["warden.user.user.session"] comes from adding timeoutable to the user model
+
+    if session["warden.user.user.session"] && Time.zone.now.to_i > timeout_timestamp.to_i
+      sign_out current_user
+      flash[:notice] = "Your session has timed out."
```

Comment:
how long does the default flash notice present itself for?

Reply:
@daryanypl Just 5 seconds.